src/tracks/playabletrack/wavetrack/ui/WaveClipAdjustBorderHandle.cpp

```diff
@@ -75,19 +75,59 @@ namespace {
     virtual void Cancel() = 0;
 
     virtual void Draw(
-        TrackPanelDrawingContext &context, 
-        const wxRect &rect, 
+        TrackPanelDrawingContext &context,
+        const wxRect &rect,
         unsigned iPass);
 
     virtual wxRect DrawingArea(
-        TrackPanelDrawingContext&, 
-        const wxRect &rect, 
-        const wxRect &panelRect, 
+        TrackPanelDrawingContext&,
+        const wxRect &rect,
+        const wxRect &panelRect,
         unsigned iPass);
  };
 
 WaveClipAdjustBorderHandle::AdjustPolicy::~AdjustPolicy() = default;
 
+namespace {
+double GetLeftAdjustLimit(
+   const WaveClip& clip, const WaveTrack& track, bool adjustingLeftBorder,
+   bool isStretchMode)
+{
+   if (!adjustingLeftBorder)
+      // A clip of one sample is invisible - who needs that ? Leave at least 2.
+      return clip.GetPlayStartTime() + 2.0 / clip.GetRate();
```

Comment:
Was this little change from a 1 sample minimum to 2 asked for by anyone?

Are you sure it always works as intended?  Maybe somewhere there is some roundoff of the inexact sample time to a discrete sample count that goes off by one.

Reply:
No, it wasn't, but in current release, if you trim so far that the visible part is only one sample and you release the trim handle, you cannot trim back (unless you Undo.) Let's ask @LWinterberg.

I think it should always work. The clip's start time is always displayed on a sample position.

Reply:
Please elaborate the comment I added at the top for the QA checklist.